### PR TITLE
editor: adjust some elements on document jsonschema

### LINKED ISSUE #739: Problems in the document editor: impossible to save
**Problem**
Sometimes, it is impossible to save a very simple document (see video).

This happen in sonardev and in production.

**Wanted improvements**

* enable to save a document

**Screen video**

https://user-images.githubusercontent.com/8154915/146376222-1d39e33a-3883-44b2-85e1-0fe1d910fe18.mp4

### LINKED ISSUE #740: Various problems in the document editor
**Describe the issue**
There are various problems in the document editor. Often it is not possible to save it

A) field `contribution` (see screenshot)
* `agent.preferred_name` should be required
* all other fields of `agent` (except `type`) should be optional
* within `identifiedBy`
    * `type` and `value` should be required
    * `source` should be required only if "type=bf:Local"
* Ideally, only `agent.preferred_name`, `role` and `affiliation` are shown by default in the editor

**To Reproduce**

Case A)
1. Create a new document
2. Add a contributor

**Expected behavior**

* Fields conditions (required, optional, etc.) are respected in the web form.
* Fields shown by default are corrected

**Software version**

Case A) observed on sonardev and sonar prod on 16.12.21

**Screenshots**
Case A)
image

### PR DESCRIPTION
**Pull request by Garfield-fr**
* Fixes the control of the availability of information on hidExpression.
* Updates the contribution field.
* Closes rero#728.
* Closes rero#739.
* Closes rero#740.

Co-Authored-by: Bertrand Zuchuat <bertrand.zuchuat@rero.ch>

## Dependency

- sonar-ui: https://github.com/rero/sonar-ui/pull/285

## Why are you opening this PR?

- To fix the problems on the editor

## How to test?

- Test all elements on document editor.
- Test all elements on deposit editor.

## Code review check list

- [ ] Commit message template compliance.
- [ ] Commit message without typos.
- [ ] File names.
- [ ] Functions names.
- [ ] Functions docstrings.
- [ ] Unnecessary commited files?
- [ ] Extracted translations?

### CODE REVIEW

**Comment by pronguen** [comment]
- [ ] for the `contribution` field
    - I should be able to add the other sub-fields (`identifiedBy`, `date_of_birth`, etc.)
    - I should be able to save, but I receive this error message:
    image

**Comment by pronguen** [comment]
- [ ] for https://github.com/rero/sonar/issues/728, the tooltip does not appear in the deposit editor. It is ok in the document editor.

File: sonar/modules/documents/jsonschemas/documents/document-v1.0.0_src.json [hunk 1]
```diff
@@ -1905,5 +1968,60 @@
...
+              "label": "contribution_role_ctb",
+              "value": "ctb"
+            }
+          ]
+        }
+      }
+    },
+    "controlledAffiliation": {
+      "title": "Controlled affiliations",
+      "type": "array",
```
Comment (jma) [inline sonar/modules/documents/jsonschemas/documents/document-v1.0.0_src.json:2016]: You do not specify the `min_items` due to editor issue, right?

**Review by jma — APPROVED** [review]

File: sonar/modules/documents/jsonschemas/documents/document-v1.0.0_src.json (hunk 1 above)
Comment (Garfield-fr) [reply sonar/modules/documents/jsonschemas/documents/document-v1.0.0_src.json:2016]: Yes, you are right

**Pull request merged**